graylog2-server/src/main/java/org/graylog2/database/suggestions/MongoEntitySuggestionService.java

```diff
@@ -101,11 +98,15 @@ public boolean hasAllPermission(final Subject subject) {
 
     public boolean hasReadPermissionForWholeCollection(final Subject subject,
                                                        final String collection) {
-        final String permission = mongoCollectionToPermission.get(collection);
-        if (permission == null) {
+        final DbEntityCatalogEntry dbEntityCatalogEntry = catalog.getByCollectionName(collection).orElse(null);
```

Comment:
I think you could rewrite this with a several map calls and falls as default, without ifs and null checks?